scripts/codestyle/run_uncrustify.sh

```diff
@@ -0,0 +1,106 @@
+#!/bin/bash
+
+if [ $# -gt 3 ]; then
+	echo "Too many parameters specified"
+	exit 1
+fi
+
+VERBOSE=false
+HASHES=()
+
+while [[ $# -gt 0 ]]
+do
+	key="$1"
+	case $key in
+	    -v|--verbose)
+	    VERBOSE=true
+	    ;;
+        -h|--help)
+		echo "run_uncrustify.sh [options] [<SHA1> <SHA2>]"
+		echo ""
+        echo "If SHA is provided to the script, uncrustify checks code style of files that are different in SHA and HEAD. "
+        echo "Otherwise uncommited diff is checked."
+        echo ""
+        echo "Options:"
+        echo "-v|--verbose : shows uncrustify suggestions for code style. When not set, only file names with differences are shown."
+        echo "-h|--help    : shows this message"
+        exit 1
+        ;;
+	    *)
+	    HASHES+=($key)
+	    ;;
+	esac
+	shift
+done
+
+if [ ${#HASHES[@]} -eq 2 ];
+then 
+	diffargs="${HASHES[0]} ${HASHES[1]}"
+fi
+
+files=$(git diff --name-only $diffargs);
+#files=$(find ../../ -type f -name "*.[ch]") #all files - I'll leave it just in case
+
+folders_to_format=$(cat folders_to_format.txt)
```

Comment:
Is the script intended to run from scripts/codestyle? It's not very convinient, could you make it run from arbitrary current directory? Useful way is to define something like `UNCRUSTIFY_BASE=$(dirname $0)` and use it as a path to folders_to_format.txt, etc

This also will allow to remove explicit processing of path (`../..`) on line 65